src/main/java/lotto/domain/WinningRank.java

```diff
@@ -34,16 +36,24 @@ public boolean isDisplay() {
         return isDisplay;
     }
 
-    public static WinningRank valueOf(int matchCount) {
+    public static WinningRank match(int matchCount, boolean matchBonus) {
+        if (isSecondMatchCondition(matchCount, matchBonus)) {
+            return SECOND;
+        }
+
         return Arrays.stream(WinningRank.values())
                 .filter(rank -> rank.matchCount == matchCount)
                 .findFirst()
                 .orElse(MISS_MATCH);
     }
 
+    private static boolean isSecondMatchCondition(int matchCount, boolean matchBonus) {
+        return matchBonus && SECOND.matchCount == matchCount;
+    }
+
     public String getStatisticsMessage(Long winningCount) {
         if (this.isDisplay()) {
-            return String.format(PRINT_STATISTICS_FORMAT,
+            return String.format(this == SECOND ? PRINT_SECOND_STATISTICS_FORMAT : PRINT_STATISTICS_FORMAT,
```

Comment:
```suggestion
            return String.format(this == SECOND ? PRINT_SECOND_STATISTICS_FORMAT : PRINT_STATISTICS_FORMAT,
```
3항 연산자 부분을 메서드로 분리하면 코드가 좀 더 읽기 쉬워질것 같아요! 🤔